Restyled `default` address block

MichelGabriel opened the pull request:
This PR adds a different style for the `default` address block in the metadata page:

- labels are added
- it now has a bordered address card
- the title is in a header
- an address card icon is added to the heading

**Screenshot of the restyled default layout**

---
<img width="729" alt="gn-default-layout" src="https://github.com/user-attachments/assets/739d223b-d669-4d2d-a7bf-c031e25f520d">
---

Furthermore, it changes the style for the `icon` layout a bit:
- the circled letter is smaller in size
- texts are now all right of the circle
- italic is changed to bold

**Screenshot of the smaller circles**

---
<img width="718" alt="gn-icon-layout" src="https://github.com/user-attachments/assets/27d45999-6797-400b-b893-b54b6f366d8d">
---


# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geonetwork/core-geonetwork/blob/main/CONTRIBUTING.md)
- [x] *Pull request* provided for `main` branch, backports managed with label
- [x] *Good housekeeping* of code, cleaning up comments, tests, and documentation
- [x] *Clean commit history* broken into understandable chucks, avoiding big commits with hundreds of files, cautious of reformatting and whitespace changes
- [x] *Clean commit message*s, longer verbose messages are encouraged
- [ ] *API Changes* are identified in commit messages
- [ ] *Testing* provided for features or enhancements using [automatic tests](https://github.com/geonetwork/core-geonetwork/blob/main/software_development/TESTING.md)
- [ ] *User documentation* provided for new features or enhancements in [manual](https://github.com/geonetwork/core-geonetwork/tree/main/docs/manual)
- [ ] *Build documentation* provided for development instructions in `README.md` files
- [ ] *Library management* using `pom.xml` dependency management. Update build documentation with intended library use and library tutorials or documentation

<!--Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.-->

<!-- If you can, it's better to give credits to organisation supporting this work:
- `Funded by NAME`
- `Funded by URL`
- `Funded by NAME URL`
-->

MichelGabriel committed:
This commit adds a different style for the `default` address block in the metadata page:
- labels are added
- it now has a bordered address card
- the title is in a header
- an address card icon is added to the heading

Furthermore, it changes the style for the `icon` layout a bit:
- the circled letter is smaller in size
- texts are now all right of the circle
- italic is changed to bold

MichelGabriel commented:
To change into the `default` layout (the `icon` layout is standard) a few view files need to be modified:

Add `default` 2 times in `src/main/resources/catalog/views/default/templates/recordView/contact.html`
```html
<div class="row" data-ng-if="mdView.current.record.contactForResource">
  <div
    data-gn-metadata-contacts="mdView.current.record.contactForResource"
    data-gn-mode="default"
    data-layout="default"
  ></div>
</div>
```

and 2 times in `src/main/resources/catalog/views/default/templates/recordView/metadatacontact.html`
```html
<div data-ng-if="mdView.current.record.contact">
  <div
    data-gn-metadata-contacts="mdView.current.record.contact"
    data-gn-mode="default"
    data-layout="default"
  ></div>
</div>
```

and make changes to these lines 
```html
    <div
      class="gn-contact-col-1"
      data-ng-if="mdView.current.record.contactForDistribution"
      data-gn-metadata-contacts="mdView.current.record.contactForDistribution"
      data-gn-mode="default"
      data-layout="icon"
    ></div>
```
in the files `src/main/resources/catalog/views/default/templates/recordView/type-dataset.html` and `src/main/resources/catalog/views/default/templates/recordView/type-series.html`

ianwallen commented:
Would you consider fixing the following in this PR?
https://github.com/geonetwork/core-geonetwork/issues/7475

MichelGabriel committed:
Add print styles to enable the printing of the address popover

MichelGabriel commented:
Added print styles for the address popover:

<img width="730" alt="gn-address-print" src="https://github.com/user-attachments/assets/4639a32f-d847-4057-86b0-1479364bb5c9">

MichelGabriel commented:
The rpint styles fix issue https://github.com/geonetwork/core-geonetwork/issues/7475

josegar74 reviewed (commented):
Some comments:

- In `icon` mode I can't get the print styles to work. If I print the page in Chrome, I get the same information as in the HTML, missing the email and address from the popup up.

- In `icon` mode, I don't understand the reason to display part of the details like email and address in the popup that seems intended to filter. I know that is not a change in the pull request, but looks odd.

- About the configuration of the `default` and `icon` mode is not the most convenient, but no need to change it. In a future pull request, could be nice to make the layout configurable in the UI settings.

ianwallen commented:
Any objection to backporting this to 4.2.x once complete?

MichelGabriel commented:
> In icon mode I can't get the print styles to work. If I print the page in Chrome, I get the same information as in the HTML, missing the email and address from the popup up.

I will have another look at the print styles

MichelGabriel committed:
Fix for the print styles, the popup style now not only works in the emulated styles but also in the print preview and print.

This commit also contains a fix for the SVG 'named circle' that was displayed too big

MichelGabriel commented:
I pushed a fix for the print styles

tylerjmchugh commented:
### Tested `default` layout and it looks good
(Also looks good in print preview / print)

![image](https://github.com/user-attachments/assets/7bf95663-7b94-42b1-8d39-bbe30cf47aa1)

### Tested print styles and it looks good

#### Default:
![image](https://github.com/user-attachments/assets/d17b4cf5-4c94-4ab2-bf80-0e76a1618eb4)

#### Emulated:
![image](https://github.com/user-attachments/assets/b222922a-940d-4e98-9774-cef11e30bdfc)

#### Print Preview / Print:
![image](https://github.com/user-attachments/assets/480c327c-11a5-437b-9bda-4ebf17cb7217)

josegar74 committed:
Fix failing unit tests.

Refactor variables used in the view and global styles.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/geonetwork/core-geonetwork?pullRequest=8289) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/geonetwork/core-geonetwork?pullRequest=8289) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: josegar74<br/>:x: MichelGabriel<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/geonetwork/core-geonetwork?pullRequest=8289) it.</sub>

josegar74 committed:
Merge pull request #14 from GeoCat/main-address-cards-test-fixes

Fix failing unit tests

geonetworkbuild commented:
The backport to `4.2.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 26377b0ea6... This commit adds a different style for the `default` address block in the metadata page:
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".
hint: Disable this message with "git config advice.mergeConflict false"

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging web-ui/src/main/resources/catalog/components/search/mdview/partials/contact.html
CONFLICT (content): Merge conflict in web-ui/src/main/resources/catalog/components/search/mdview/partials/contact.html
Auto-merging web-ui/src/main/resources/catalog/locales/en-v4.json
CONFLICT (content): Merge conflict in web-ui/src/main/resources/catalog/locales/en-v4.json
Auto-merging web-ui/src/main/resources/catalog/style/gn_metadata.less
Auto-merging web-ui/src/main/resources/catalog/views/default/less/gn_result_default.less

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-4.2.x 4.2.x
# Navigate to the new working tree
cd .worktrees/backport-4.2.x
# Create a new branch
git switch --create backport-8289-to-4.2.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 26377b0ea657eda58190780301a149816cbc0c68,db79b520d4358b568fa6f7660ce87088dbd328d8,490cd8a506a99bfad5745768201d21dadccc5043,d653aacc17107f02c0febdf756dcfcd305f3a3fe,925ec382e3a492fe97a72c844b2474546b129620
# Push it to GitHub
git push --set-upstream origin backport-8289-to-4.2.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-4.2.x
```
Then, create a pull request where the `base` branch is `4.2.x` and the `compare`/`head` branch is `backport-8289-to-4.2.x`.
